webboot/webboot.go

```diff
@@ -23,9 +40,88 @@ var (
 	ipv6    = flag.Bool("ipv6", true, "use IPV6")
 )
 
+func wgetArgs(argURL string) (read io.ReadCloser, outFile string, werr error) {
+	//checks if argURL is a key in osSystems map
+	if _, ok := osSystems[argURL]; !ok {
+		//if argURL is not the map, use it as a URL link for wget
+		read, werr = wget(argURL)
+	} else {
+		// if argURL is in osSystems map, use its download link for wget
+		read, werr = wget(osSystems[argURL].DownloadLink)
+		outFile = argURL
+	}
+	return read, outFile, werr
+}
+
+func wget(arg string) (io.ReadCloser, error) {
+	resp, err := http.Get(arg)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+func writer(read io.ReadCloser, fileName string) (string, error) {
+	w, err := os.Create(fileName)
+	if err != nil {
+		return "", err
+	}
+
+	defer w.Close()
+
+	_, err = io.Copy(w, read)
+	return fileName, err
+}
+
+func outputName(argURL string) (outFile string) {
+	//if output file is not initialized, argURl is parsed to generate a name for the output file
+	url, err := url.Parse(argURL)
+
+	if err != nil {
+		log.Fatalf("Error parsing the URL, got %v, want nil", err)
+	}
+	// takes the last part of the url Path as the name of output file
+	if url.Path != "" && url.Path[len(url.Path)-1] != '/' {
+		outFile = path.Base(url.Path)
+	} else {
+		outFile = "index.html"
+	}
+	return outFile
+}
+
+func usage() {
+	log.Printf("Usage: %s [ARGS] URL\n", os.Args[0])
+	flag.PrintDefaults()
```

Comment:
by convention, usage should exit, so after flag.PrintDefaults you can just
os.Exit(1)

Reply:
I originally had os.Exit() in the usage() function, but then I did not know if I should exit the user with simply a return statement inside of main function. I will revert it back to the way I had it with the "os.Exit(1)"